### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by DmytroBondariev**

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -1,34 +1,62 @@
...
+                self.decks.append(Deck(row, column))
 
-    def get_deck(self, row, column):
-        # Find the corresponding deck in the list
-        pass
+    def get_deck(
+            self,
+            row: int,
+            column: int
+    ) -> Deck:
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: It returns Optional[Deck]

File: app/main.py [hunk 2]
```diff
@@ -1,34 +1,62 @@
...
-    def fire(self, row, column):
-        # Change the `is_alive` status of the deck
-        # And update the `is_drowned` value if it's needed
-        pass
+    def fire(
+            self,
+            row: int,
+            column: int
+    ) -> None:
+        self.get_deck(row, column).is_alive = False
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: get_deck can return None

File: app/main.py [hunk 3]
```diff
@@ -1,34 +1,62 @@
...
+        for deck in self.decks:
+            self.is_drowned = True
+            if deck.is_alive:
+                self.is_drowned = False
+                break
```
Comment (ArsenPidhoretskyi) [inline app/main.py]: Use any or all

Comment (ArsenPidhoretskyi) [reply app/main.py]: `self.is_drowned = not any(deck.is_alive for deck in self.decks)`

File: app/main.py [hunk 4]
```diff
@@ -1,34 +1,62 @@
...
-    def __init__(self, ships):
-        # Create a dict `self.field`.
-        # Its keys are tuples - the coordinates of the non-empty cells,
-        # A value for each cell is a reference to the ship
-        # which is located in it
-        pass
+    def __init__(self, ships: list) -> None:
+        self.field = dict()
+        for ship in ships:
+            battleship = Ship(ship[0], ship[1])
```
Comment (ArsenPidhoretskyi) [inline app/main.py:54]: is battleship clear???

**Review by ArsenPidhoretskyi — CHANGES_REQUESTED** [review]

File: app/main.py [hunk 5]
```diff
@@ -1,34 +1,64 @@
...
-        # which is located in it
-        pass
-
-    def fire(self, location: tuple):
-        # This function should check whether the location
-        # is a key in the `self.field`
-        # If it is, then it should check if this cell is the last alive
-        # in the ship or not.
-        pass
+    def __init__(self, ships: list) -> None:
```
Comment (vasylhnatiuk) [inline app/main.py]:
```suggestion
    def __init__(self, ships: list[tuple]) -> None:
```

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

**Review by vasylhnatiuk — APPROVED** [review]

**Pull request closed without merging**